### PR TITLE
switch to dune

### PR DESCRIPTION
**Pull request by redianthus**
Hi,

I just switched the project to dune.

Cheers.

### CODE REVIEW

File: gencode/main.ml [hunk 1]
```diff
@@ -100,7 +98,7 @@ let program =
   Cohttp_lwt_unix.Client.get
     ("http://www.unicode.org/emoji/charts/emoji-list.html"
      |> Uri.of_string) >>= fun (_, body) ->
-  Cohttp_lwt_body.to_string body >>= fun html ->
+  Cohttp_lwt__Body.to_string body >>= fun html ->
```
Comment (fxfactorial) [inline gencode/main.ml:101]: PR says switch to dune but this is code changes. What happened in upstream cohttp ?

Comment (redianthus) [reply gencode/main.ml:101]: I don't know. I've got latest cohttp installed, there wasn't any version constraint here so I just fixed it.

Comment (fxfactorial) [reply gencode/main.ml:101]: Did you test it, that is, everything stayed the same yes? Even after the update of library

Comment (redianthus) [reply gencode/main.ml:101]: I added a commit with a test which seems to work. You can run it with `dune exec test/test.exe`.

**Pull request merged**